src/azure-cli-testsdk/azure/cli/testsdk/utilities.py

```diff
@@ -182,26 +182,33 @@ def process_response(self, response):
         return response
 
 
-class MSGraphUserReplacer(RecordingProcessor):
-    def __init__(self, test_user, mock_user):
-        self.test_user = test_user
-        self.mock_user = mock_user
+class MSGraphNameReplacer(RecordingProcessor):
+    """If a name appears in URL, it should be percent-encoded, e.g.
+      - test-name+/ is encoded as test-name%2B%2F
+      - example@example.com is encoded as example%40example.com
+
+    Theoretically, GeneralNameReplacer should also follow this pattern, but since we haven't seen any ARM name that
+    is percent-encoded, we only replace percent-encoded names for MS Graph for better test performance.
```

Comment:
I tried to change `GeneralNameReplacer`'s logic to also replace encoded names, but it turns out to be an overdo since ARM resource names usually don't need to be encoded, so we only do it specially for MS Graph.

Reply:
Do we have test to cover the changes?

Reply:
`test_special_characters` is exactly for that purpose.